doc/arkode/guide/source/Usage/ASA.rst

```diff
@@ -0,0 +1,159 @@
+
+#. *Initialize parallel or multi-threaded environment, if appropriate.*
+
+#. *Create the SUNDIALS simulation context object*
+
+#. *Set problem dimensions, etc.*
+
+#. *Set vector of initial values*
+
+#. *Create a stepper object*
+
+#. *Set optional inputs*
+
+#. *Specify rootfinding problem*
+
+#. Create a :c:type:`SUNAdjointCheckpointScheme` object
+
+   Create the :c:type:`SUNAdjointCheckpointScheme` object by calling ``SUNAdjointCheckpointScheme_Create_*``.
+   Available :c:type:`SUNAdjointCheckpointScheme` implementations are found in
+   section :numref:`SUNAdjoint.CheckpointScheme`.
+   
+#. Attach the checkpoint scheme object to ARKODE 
+   
+   Call :c:func:`ARKodeSetAdjointCheckpointScheme()`.
+
+#. *Advance solution in time*
+
+#. *Get optional outputs*
+
+#. Create the sensitivities vector with the terminal condition
+
+   The sensitivities vector must be an instance of the :ref:`ManyVector N_Vector implementation <NVectors.ManyVector>`.
+   You will have one subvector for the initial condition sensitivities and an additional subvector if you
+   want sensitivities with respect to parameters.
+   The vector should contain the terminal condition from the forward problem.
+
+#. Create the :c:type:`SUNAdjointStepper` object
+
+   Call :c:func:`ERKStepCreateAdjointStepper` or :c:func:`ARKStepCreateAdjointStepper`.
+
+#. Set either the Jacobian-vector product, vector-Jacobian product, or Jacobian evaluation functions
+
+   Users must supply one of:
+   
+   * :math:`(\partial f/\partial y)^*v`,
+   * :math:`v^*(\partial f/\partial y)`,
+   * :math:`(\partial f/\partial y)`,
+
+   and, if sensitivities with respect to the parameters is desired, one of
+
+   * :math:`(\partial f/\partial p)^*v`,
+   * :math:`v^*(\partial f/\partial p)`,
+   * :math:`(\partial f/\partial p)`.
+
+   These user-supplied routines can be set with :c:func:`SUNAdjointStepper_SetJacHermitianTransposeVecFn`, or
+   :c:func:`SUNAdjointStepper_SetJacFn`.
+
+#. Set optional ASA input
+
+   Refer to :numref:`SUNAdjoint.Stepper` for options.
+
+#. Advance the adjoint sensitivity analysis ODE
+
+   Call :c:func:`SUNAdjointStepper_Evolve`.
+
+#. Get optional ASA outputs
+
+   Refer to :numref:`SUNAdjoint.Stepper` for options.
+
+#. Deallocate memory for ASA objects
+
+   Deallocate the sensitivities vector, :c:type:`SUNAdjointStepper`, 
+   and :c:type:`SUNAdjointCheckpointScheme` objects.
+
+#. *Deallocate memory for solution vector*
+
+#. Free solver memory
+
+   * If an ARKODE stepper module was used as a partition IVP integrator, call
+     :c:func:`SUNStepper_Destroy` and :c:func:`ARKodeFree` to free the memory
+     allocated for that integrator.
+
+   * If a user-defined partition integrator was supplied, free the integrator
+     content and call :c:func:`SUNStepper_Destroy` to free the :c:type:`SUNStepper`
+     object.
+
+   * Call :c:func:`ARKodeFree` to free the memory allocated for the
+     SplittingStep integration object.
```

Comment:
The instruction at the top of this skeleton only explained how to create the SUNStepper using ARKODE (via :c:func:`ERKStepCreateAdjointStepper` or :c:func:`ARKStepCreateAdjointStepper`), so this second bullet lacks context.  Furthermore, I don't see why the third bullet is here at all.  Lastly, nothing about this skeleton discusses partition integrators -- I think these may be the result of a bad merge?
```suggestion
   * Call :c:func:`SUNStepper_Destroy` and :c:func:`ARKodeFree` to free the memory
     allocated for the SUNStepper and ARKODE integrator objects.
```

Reply:
Yes, this is just from a bad merge / copy-paste.